x/evm/keeper/erc20.go

```diff
@@ -189,19 +190,18 @@ func (k Keeper) CallContractWithInput(
 	}
 
 	unusedBigInt := big.NewInt(0)
-	evmMsg := gethcore.NewMessage(
-		fromAcc,
-		contract,
-		nonce,
-		unusedBigInt, // amount
-		gasLimit,
-		unusedBigInt, // gasFeeCap
-		unusedBigInt, // gasTipCap
-		unusedBigInt, // gasPrice
-		contractInput,
-		gethcore.AccessList{},
-		!commit, // isFake
-	)
+	evmMsg := &core.Message{
```

Comment:
I love that this constructor is a struct now instead of a function.  Much easier to read what the fields mean

Reply:
Yes, I prefer literal constructors over constructors in Golang (and in general) because you're never sure if the constructors perform any side effects.